astronomer/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -0,0 +1,256 @@
+            return None
+        else:
+            await self.wait_for_job(self.job_id)
+            await self.check_job_success(self.job_id)
+            success_msg = "AWS Batch job (%s) succeeded" % self.job_id
+            self.log.info(success_msg)
+            return {"status": "success", "message": success_msg}
+
+    async def check_job_success(self, job_id: str) -> bool:  # type: ignore[override]
+        """
+        Check the final status of the Batch job; return True if the job
+        'SUCCEEDED', else raise an AirflowException
+
+        :param job_id: a Batch job ID
+
+        :raises: AirflowException
+        """
+        job = await self.get_job_description(job_id)
+        job_status = job.get("status")
+        if job_status == self.SUCCESS_STATE:
+            self.log.info("AWS Batch job (%s) succeeded: %s", job_id, job)
+            return True
+
+        if job_status == self.FAILURE_STATE:
+            raise AirflowException(f"AWS Batch job ({job_id}) failed: {job}")
+
+        if job_status in self.INTERMEDIATE_STATES:
+            raise AirflowException(f"AWS Batch job ({job_id}) is not complete: {job}")
+
+        raise AirflowException(f"AWS Batch job ({job_id}) has unknown status: {job}")
+
+    @staticmethod
+    async def delay(delay: Union[int, float, None] = None) -> None:  # type: ignore[override]
+        """
+        Pause execution for ``delay`` seconds.
+
+        :param delay: a delay to pause execution using ``time.sleep(delay)``;
+            a small 1 second jitter is applied to the delay.
+
+        .. note::
+            This method uses a default random delay, i.e.
+            ``random.uniform(DEFAULT_DELAY_MIN, DEFAULT_DELAY_MAX)``;
+            using a random interval helps to avoid AWS API throttle limits
+            when many concurrent tasks request job-descriptions.
+        """
+        if delay is None:
+            """Using random.uniform() causes Bandit security check to fail with Issue: [B311:blacklist] Standard
+            pseudo-random generators are not suitable for security/cryptographic purposes. Hence,
+            using random.sample() instead"""
+            delay = sample(
+                list(range(BatchClientHookAsync.DEFAULT_DELAY_MIN, BatchClientHookAsync.DEFAULT_DELAY_MAX)), 1
+            )[0]
+        else:
+            delay = BatchClientHookAsync.add_jitter(delay)
+        await asyncio.sleep(delay)
+
+    async def wait_for_job(self, job_id: str, delay: Union[int, float, None] = None) -> None:  # type: ignore[override]
+        """
+        Wait for Batch job to complete
+
+        :param job_id: a Batch job ID
+
+        :param delay: a delay before polling for job status
+
+        :raises: AirflowException
+        """
+        await self.delay(delay)
+        await self.poll_for_job_running(job_id, delay)
+        await self.poll_for_job_complete(job_id, delay)
+        self.log.info("AWS Batch job (%s) has completed", job_id)
```

Comment:
Is this the expected behaviour, if the job succeeds or fails then airflow task will be considered a success

Reply:
`poll_for_job_running` and `poll_for_job_complete`  call `poll_job_status`  where `AirflowException` is raised.Dont you think this is sufficient ?

Reply:
Thanks for confirming it. I was just concerned that both SUCCEEDED or FAILED will be considered as completed or not.